pkg/liqonet/ipam/ipamStorage.go

```diff
@@ -54,20 +56,22 @@ type IpamStorage interface {
 	updatePodCIDR(podCIDR string) error
 	updateServiceCIDR(serviceCIDR string) error
 	updateNatMappingsConfigured(natMappingsConfigured map[string]netv1alpha1.ConfiguredCluster) error
-	getClusterSubnets() (map[string]netv1alpha1.Subnets, error)
-	getPools() ([]string, error)
-	getExternalCIDR() (string, error)
-	getEndpointMappings() (map[string]netv1alpha1.EndpointMapping, error)
-	getPodCIDR() (string, error)
-	getServiceCIDR() (string, error)
-	getNatMappingsConfigured() (map[string]netv1alpha1.ConfiguredCluster, error)
+	getClusterSubnets() map[string]netv1alpha1.Subnets
+	getPools() []string
+	getExternalCIDR() string
+	getEndpointMappings() map[string]netv1alpha1.EndpointMapping
+	getPodCIDR() string
+	getServiceCIDR() string
+	getNatMappingsConfigured() map[string]netv1alpha1.ConfiguredCluster
 	goipam.Storage
 }
 
 // IPAMStorage is an implementation of IpamStorage that takes advantage of the CRD IpamStorage.
 type IPAMStorage struct {
-	dynClient    dynamic.Interface
-	resourceName string
+	sync.RWMutex
```

Comment:
In #889 we introduced a mutex to prevent to solve concurrency problems. The mutex introduced in this PR narrows the critical section to less code than the previous one. We should remove the old one.

Reply:
I'm not that confident that this mutex would solve the problem, since no additional guarantee is introduced with respect to the previous implementation (it just ensures the cached values is not modified while being read).